### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by izaiasneto4**

### CODE REVIEW

File: methods/my_each.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    if block_given?
+      i = 0
+      while i < size
+        yield at(i)
+        i += 1
+      end
+    else
+      'Error: my_each must have a block'
+    end
+  end
```
Comment (maelfosso) [inline methods/my_each.rb]: Should return an enumerator
```ruby
irb(main):017:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):018:0> array.each
=> #<Enumerator: [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]:each>
irb(main):019:0> array.my_each
=> "Error: my_each must have a block"
irb(main):020:0> 
```

File: methods/my_each_with_index.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      index = 0
+      while i < size
+        yield(index, at(i))
+        index += 1
+        i += 1
+      end
+    else
+      'Error: my_each_with_index must have a block'
+    end
+  end
```
Comment (maelfosso) [inline methods/my_each_with_index.rb]: Should return an **enumerator**
```ruby
irb(main):022:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):023:0> array.each_with_index
=> #<Enumerator: [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]:each_with_index>
irb(main):024:0> array.my_each_with_index
=> "Error: my_each_with_index must have a block"
irb(main):025:0> 
```

File: methods/my_select.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  def my_select
+    raise 'Error: No block was given' unless block_given?
+
+    final_arr = []
+    my_each do |i|
+      final_arr.push(i) if yield(i)
+    end
+
+    final_arr
+  end
```
Comment (maelfosso) [inline methods/my_select.rb]: Should return an enumerator
```ruby
irb(main):027:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):028:0> array.select
=> #<Enumerator: [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]:select>
irb(main):029:0> array.my_select
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):29
        1: from main.rb:67:in `my_select'
RuntimeError (Error: No block was given)
irb(main):030:0> 
```

File: methods/my_all?.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return false unless i }
+    else
+      my_each { |i| return false unless i.is_a?(arg) }
+    end
+    true
+  end
```
Comment (maelfosso) [inline methods/my_all?.rb]:
```ruby
irb(main):033:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):034:0> words.all?(/d/)
=> true
irb(main):035:0> words.my_all?(/d/)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):35
        4: from main.rb:39:in `my_all?'
        3: from main.rb:11:in `my_each'
        2: from main.rb:39:in `block in my_all?'
        1: from main.rb:39:in `is_a?'
TypeError (class or module required)
irb(main):036:0> 
```

Comment (maelfosso) [reply methods/my_all?.rb]:
```ruby
irb(main):038:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):039:0> array.all?(3)
=> false
irb(main):040:0> array.my_all?(3)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):40
        4: from main.rb:39:in `my_all?'
        3: from main.rb:11:in `my_each'
        2: from main.rb:39:in `block in my_all?'
        1: from main.rb:39:in `is_a?'
TypeError (class or module required)
irb(main):041:0>
```

File: methods/my_any?.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |i| return true if yield(i) }
+    elsif arg.nil?
+      my_each { |i| return true if i }
+    else
+      my_each { |i| return true if i.is_a?(arg) }
+    end
+    false
+  end
```
Comment (maelfosso) [inline methods/my_any?.rb]:
```ruby
irb(main):043:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):044:0> words.any?(/z/)
=> false
irb(main):045:0> words.my_any?(/z/)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):45
        4: from main.rb:50:in `my_any?'
        3: from main.rb:11:in `my_each'
        2: from main.rb:50:in `block in my_any?'
        1: from main.rb:50:in `is_a?'
TypeError (class or module required)
irb(main):046:0> 
```

Comment (maelfosso) [reply methods/my_any?.rb]:
```ruby
irb(main):048:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):049:0> words[0] = 'cat'
=> "cat"
irb(main):050:0> words.any?('cat')
=> true
irb(main):051:0> words.my_any?('cat')
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):51
        4: from main.rb:50:in `my_any?'
        3: from main.rb:11:in `my_each'
        2: from main.rb:50:in `block in my_any?'
        1: from main.rb:50:in `is_a?'
TypeError (class or module required)
irb(main):052:0>
```

File: methods/my_none?.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |i| return false if yield(i) }
+    elsif arg.nil?
+      my_each { |i| return false if i }
+    else
+      my_each { |i| return false if i.is_a(arg) }
+    end
+    true
+  end
```
Comment (maelfosso) [inline methods/my_none?.rb]:
```ruby
irb(main):054:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):055:0> array.none?(String)
=> true
irb(main):056:0> array.my_none?(String)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):56
        3: from main.rb:61:in `my_none?'
        2: from main.rb:11:in `my_each'
        1: from main.rb:61:in `block in my_none?'
NoMethodError (undefined method `is_a' for 5:Integer)
Did you mean?  is_a?
```

Comment (maelfosso) [reply methods/my_none?.rb]:
```ruby
irb(main):061:0> words 
=> ["cat", "door", "rod", "blade"]
irb(main):062:0> words[0] = 5
=> 5
irb(main):063:0> words.none?(5)
=> false
irb(main):064:0> words.my_none?(5)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):64
        4: from main.rb:61:in `my_none?'
        3: from main.rb:11:in `my_each'
        2: from main.rb:61:in `block in my_none?'
        1: from main.rb:61:in `is_a?'
TypeError (class or module required)
irb(main):065:0>
```

File: methods/my_count.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+  def my_count
+    count = 0
+
+    my_each do
+      count += 1
+    end
+
+    count
+  end
```
Comment (maelfosso) [inline methods/my_count.rb]:
```ruby
irb(main):067:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):068:0> array.count(3)
=> 7
irb(main):069:0> array.my_count(3)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):69
        1: from main.rb:77:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):070:0> 
```

Comment (maelfosso) [reply methods/my_count.rb]:
```ruby
irb(main):072:0> block = proc { |num| num < 9/2 }
=> #<Proc:0x0000556516922310@(irb):72>
irb(main):073:0> array.count(&block)
=> 37
irb(main):074:0> array.my_count(&block)
=> 100
irb(main):075:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):076:0> 
```

File: methods/my_map.rb [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    raise 'Error: No block was given' unless block_given?
+
+    final_arr = []
+
+    my_each do |i|
+      final_arr.push(yield(i))
+    end
+
+    final_arr
+  end
```
Comment (maelfosso) [inline methods/my_map.rb]:
```ruby
irb(main):079:0> array
=> [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]
irb(main):080:0> array.map
=> #<Enumerator: [5, 1, 5, 5, 6, 3, 3, 3, 7, 5, 7, 5, 5, 7, 0, 6, 1, 2, 5, 5, 6, 4, 6, 7, 4, 7, 7, 6, 5, 7, 5, 4, 6, 2, 4, 2, 6, 3, 7, 0, 1, 2, 6, 6, 5, 6, 6, 5, 0, 8, 8, 7, 0, 6, 0, 5, 6, 3, 2, 6, 4, 3, 7, 2, 4, 8, 6, 0, 8, 8, 3, 1, 1, 0, 4, 8, 7, 2, 6, 0, 7, 2, 0, 6, 1, 2, 7, 6, 1, 0, 7, 7, 2, 6, 5, 0, 1, 0, 8, 6]:map>
irb(main):081:0> array.my_map
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):81
        1: from main.rb:88:in `my_map'
RuntimeError (Error: No block was given)
irb(main):082:0>
```

**Comment by Ispirett** [comment]
### please use the correct fit work flow ` feature-branch - > development` and create another pull request.

**Pull request closed without merging**